diff --git a/bfc b/bfc
index 5080dc7..5e44707 100755
--- a/bfc
+++ b/bfc
@@ -152,13 +152,12 @@ sub build {
         }
     }
     $buildopts{stage} = 'a' unless $buildopts{stage};
-    print "$buildopts{stage}\n";
     $buildopts{options} = '--short-circuit' if $opts{o};
     my $builder = Youri::Package::RPM::Builder->new(
         sourcedir   => cwd(),
         verbose     => 2
     );
-    $builder->build(find_spec(), %buildopts);
+    eval { $builder->build(find_spec(), %buildopts) };
 }
 
 =head2 cleantree
@@ -310,7 +309,7 @@ sub sync {
                 }
             } else {
                 delete $abfyml->{sources}{$file};
-                print "$file removed from .abf.yml";
+                print "$file removed from .abf.yml\n";
             }
         }
     }
@@ -359,7 +358,7 @@ sub sync {
     }
 
     if (@filestoupload) {
-        print "\nYou need to upload following files to file store:\n";
+        print "\nFiles to be uploaded to file store:\n";
         print join("\n", @filestoupload) . "\n";
     }
 }
@@ -458,7 +457,7 @@ sub fs_upload {
     my @results;
     foreach (@_) {
         my ($filename) = m{([^/]+)$};
-        my $url = "$baseurl/api/v1/file_stores";
+        my $url = "$baseurl/api/v1/upload";
         $HTTP::Request::Common::DYNAMIC_FILE_UPLOAD = 1;
         my $request = POST $url,
                 Content_Type    => 'form-data',
